docs/notes/2.23.x.md

```diff
@@ -90,6 +90,8 @@ Exported virtualenvs can use Pants-provided Python if a `PythonProvider` backend
 
 The docs for the `check` goal have been updated to state that third-party type stubs must be installed in the same resolve as the code.
 
+When building `python_google_cloud_function` targets, the `complete_platforms` field may now be specified as either a `file` target or a `resource` target.
```

Comment:
I think this is a change to the core pex utility fields/rules, and thus this applies to all targets that use `PexCompletePlatformsField`, including `pex_binary`, `python_aws_lambda_function`, `python_aws_lambda_layer`, and potentially others?

So maybe these notes could be adjusted.

Reply:
I agree about the other FaaS use cases like `python_aws_lambda_function` and friends and can include those in the notes. However, I don't think `pex_binary` uses the rule that I modified because it already worked when specifying `complete_platforms` as a resource target (mentioned in the linked issue, and I was able to confirm this manually as well).

Reply:
Oh, hm! Good catch...

I don't doubt your tests... but I'm confused, because it seems like packaging a `pex_binary` flows through these rules:

1. https://github.com/pantsbuild/pants/blob/cfb3cafcac2386c909bc4a3fee3997e1bd8b03d7/src/python/pants/backend/python/goals/package_pex_binary.py#L147-L149

2. https://github.com/pantsbuild/pants/blob/cfb3cafcac2386c909bc4a3fee3997e1bd8b03d7/src/python/pants/backend/python/util_rules/pex.py#L149-L155

3. https://github.com/pantsbuild/pants/blob/cfb3cafcac2386c909bc4a3fee3997e1bd8b03d7/src/python/pants/backend/python/util_rules/pex.py#L127-L146

That is, it does end up in the rule modified in this PR 🤔